src/journal/src/Journal/Internal/ByteBuffer.hs

```diff
@@ -257,6 +266,36 @@ putBytes src dest = do
 getBytes :: ByteBuffer -> Int -> Int -> IO [Word8]
 getBytes bb offset len = undefined
 
+putByteString :: ByteBuffer -> BS.ByteString -> IO ()
+putByteString bb bs = do
+  let (fptr, I# offset#, I# len#) = BS.toForeignPtr bs
+  boundCheck bb (I# (len# -# 1#))
+  withForeignPtr fptr $ \(Ptr addr#) -> IO $ \s ->
+    case copyAddrToByteArray# addr# (bbData bb) offset# len# s of
+      s' -> (# s', () #)
+
+putLazyByteString :: ByteBuffer -> LBS.ByteString -> IO ()
+putLazyByteString bb lbs = do
+  let (fptr, I# offset#, I# len#) = BS.toForeignPtr (LBS.toStrict lbs)
+  boundCheck bb (I# (len# -# 1#))
+  withForeignPtr fptr $ \(Ptr addr#) -> IO $ \s ->
+    case copyAddrToByteArray# addr# (bbData bb) offset# len# s of
+      s' -> (# s', () #)
+
+getByteString :: ByteBuffer -> Int -> IO BS.ByteString
+getByteString bb len = do
+  bytes <- replicateM len (getByte bb)
+  return (BS.packBytes bytes)
+
+getLazyByteString :: ByteBuffer -> Int -> IO LBS.ByteString
+getLazyByteString bb len = do
+  bytes <- replicateM len (getByte bb)
+  return (LBS.packBytes bytes)
```

Comment:
So we can't really make this lazy, so we might as well use the strict version and convert, (It shouldn't really affect performance) so when we optimize the strict version we get this one for free.
```suggestion
getLazyByteString :: ByteBuffer -> Int -> IO LBS.ByteString
getLazyByteString bb len = do
  bs <- getByteString bb len
  return (LBS.fromStrict bs)
```

A `LBS.ByteString` is basically just a `[BS.ByteString]` and `LBS.fromStrict` looks like this:
```haskell
-- |/O(1)/ Convert a strict 'ByteString' into a lazy 'ByteString'.
fromStrict :: BS.ByteString -> ByteString
fromStrict (BS.BS _ 0) = Empty
fromStrict bs = Chunk bs Empty
```

`LBS.packBytes` would make it a list of 32 bytes long `BS.ByteString`.